### ISSUE TITLE
Checkpoint Challenge Required on a Linux Server

### ISSUE THREAD

**Issue by Luc1412** [ping/instagram_private_api#178]
## Please follow the guide below

- Issues submitted without this template format will be **ignored**.
- Please read the questions **carefully** and answer completely.
- Do not post screenshots of error messages or code.
- Put an `x` into all the boxes [ ] relevant to your issue (==> [x] *no* spaces).
- Use the *Preview* tab to see how your issue will actually look like.
- Issues about reverse engineering is out of scope and will be closed without response.
- Any mention of spam-like actions or spam-related tools/libs/etc is strictly **not allowed**.

---

### Before submitting an issue, make sure you have:
- [x] Updated to the lastest version v1.6.0
- [x] Read the [README](https://github.com/ping/instagram_private_api/blob/master/README.md) and [docs](https://instagram-private-api.readthedocs.io/en/latest/)
- [x] [Searched](https://github.com/ping/instagram_private_api/search?type=Issues) the bugtracker for similar issues including **closed** ones
- [x] Reviewed the sample code in [tests](https://github.com/ping/instagram_private_api/tree/master/tests) and [examples](https://github.com/ping/instagram_private_api/tree/master/examples)

### Which client are you using?
- [x] app (``instagram_private_api/``)
- [ ] web (``instagram_web_api/``)

---

### Describe your issue

I currently use the API to automate Gameupdates/news by publishing automatically images.

Some days ago I got the Checkpoint Challenge Required on my Debian 9 Server which prevent me to use the API. I already read issues which say that I need to manually login into Instagram on the Device. 

The Problem: I got a Linux KVM which I controll via SSH with no way to get a graphical interface/open Chrome.

Is there a way to solve the "challenge" only via SSH access or another device?

**Comment by wlerin** [comment]
It doesn't need to be on the same device.

**Comment by martjushev** [comment]
I have the same problem. Solved the challenge on my local machine (was even using the same proxy) - does not help, a new one appears on server.

**Comment by Luc1412** [comment]
I think for me it's another problem. The "checkpoint challenge" was no Captcha for me. My account got softbanned (Some features are disabled like following) for short time.

**Comment by KreditorHD** [comment]
Ok guys. So I am working with Instagram marketing and I read in a quite big forum, that Instagram in softbanning and followblocking accounts that overreach their new limits. So that might be the Problem here.

Here is an example of someone in the forum:

> I’ve seen a few theories.
> 
> I’d just like to put my info out there.
> No Blocks on this account:
> 
> Old account - over 6000 actions last 30 days. Added to Jarvee May 23rd
> 4G proxy
> Yesterday did over 600 actions
> Today doing unfollows - at about 200 so far today.
> Account does not meet the “new only” or “6000 action criteria” mentioned as theories.
> Account is used manually a lot on top of auto. Manual…
> DM’s, comments, stories on top of that. Even the odd report or two put in.
> No issues so far at all.
> Account around 12 months old. Has had two copywrite notices (both contested) and a recent SB - out of shadowban now.

So its a most likely a problem by Instagram and not the API or your bot itself

**Comment by tomquirk** [comment]
I'm getting the challenge and it's my first time using this project.

I got the exception, then checked instagram.com. I got a 6-digit code sent to my email, which I completed, then tried the API again. Got the exception again, and then on checking instagram.com again was forced to change my password.

Think it's the API!

**Comment by wlerin** [comment]
I got softbanned just using the website to follow a bunch of accounts, so I'm with @KreditorHD thinking this is an issue with Instagram rather than the API specifically.

**Comment by ehoppmann** [comment]
I have the same issue. Solving the challenge in a web browser on the same machine doesn't help (I tried a few times as tomquirk mentioned above).

FYI, I believe that this project implements interactive instagram challenge handling, so that may be helpful if you decide to try to implement something: https://github.com/rarcega/instagram-scraper

**Comment by qwertyNodes** [comment]
same problem here, but seems that this guyes solved this issue
https://github.com/rarcega/instagram-scraper

maybe is possible implement same here ?
hope this issue will be solved

**Comment by qwertyNodes** [comment]
> I have the same issue. Solving the challenge in a web browser on the same machine doesn't help (I tried a few times as tomquirk mentioned above).
> 
> FYI, I believe that this project implements interactive instagram challenge handling, so that may be helpful if you decide to try to implement something: https://github.com/rarcega/instagram-scraper

same issue here, this guyes solved this
https://github.com/rarcega/instagram-scraper

maybe is possible implement same here ?
hope this issue will be solved

**Comment by tomquirk** [comment]
yea, the problem we'll need to solve is getting the challenge in the first place; the authed web api works fine, just the private/app api

**Comment by moodiest** [comment]
Any advancements on tackling this issue?

I've tried logging in on my phone and chrome but there is no challenge present, it's just when using the PyInstaLive script which is this that I get a challenge.

**Comment by Luc1412** [comment]
For me, it works fine. My Account got temporarily blocked by IG. Some time ago all the features got reenabled. The only thing what was confusing that it showed the that there is a problem with the checkpoint instead of the ban.

**Comment by silau2005** [comment]
I like this library and use it sometimes so I would like to share my 2 cents.

Thanks ehoppman, qwertyNodes and tomquirk for providing some good insights into the topic. The idea of creating an interactive solution to handle checkpoint challenge is quite good. Here is my flow and the related [code](https://poorlau.com/blog/instagram-checkpoint-challenge-required/):

1. Override login method (in endpoints/accounts)
2. Expose part of the _call_api method (in client)
3. Capture the checkpoint challenge exception
4. Handle the exception by mimic Instagram Scraper's login_challenge method

**Comment by ping** [comment]
https://instagram-private-api.readthedocs.io/en/latest/faq.html#what-does-checkpoint-challenge-required-challenge-required-mean

There is no support for any IG challenges.
